public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -313,6 +313,16 @@ export function requestSupportsSplitting(allQueries: LokiQuery[]) {
   return queries.length > 0;
 }
 
+export function requestSupportsSharding(allQueries: LokiQuery[]) {
+  const queries = allQueries
+    .filter((query) => !query.hide)
+    .filter((query) => !query.refId.includes('do-not-shard'))
+    .filter((query) => query.expr)
+    .filter((query) => query.direction === LokiQueryDirection.Scan || query.refId?.startsWith('log-volume-'));
```

Comment:
You can use `REF_ID_STARTER_LOG_VOLUME` for `'log-volume-'`

Reply:
I am torn to decide if we should check that `query.expr` is only a `sum by()` expression. Theoretically users can just create queries with `log-volume-` ref ids manually and if those are not `sum by()` they would break by this change.

Reply:
We can ditch the refId check and test for sum by() to decide whether to shard or not. What do you think of that?